src/features/read/readSlice.ts

```diff
@@ -178,9 +179,51 @@ const selectQuestion = (state: RootState): IQuestion | null => {
 
 export const selectMode = (state: RootState): Mode | null => {
   const treatment = selectTreatment(state);
+  const book = selectBook(state)
+  const bookType = book?.fields.type;
+
+  //check if pre- or post-test
+  if( bookType === "pre-test" || bookType === "post-test" ){
+  //make sure book isn't null (makes getPage feel better)
+    if(book){
+      //if sys.id is a question, then continue. Otherwise, it's narrative page so skip over
+      if(Book.getPage(book,selectPageNumber(state))?.sys.contentType.sys.id === "question"){
+        //since it's a question, grab it
+        const question = selectQuestion(state);
+        
+        if (!question) {
+            throw new Error(
+              `Missing question. Current questions is of type null.`
+            );
+        }
+    
+        //try to get numberPrompt, if we can't, then it's a sizePrompt
+        try{
+          if(Question.getPrompt(question,"number")){
+            return "number";
```

Comment:
This try-catch approach works, but I think it can be improved. If an error gets thrown for a different reason than we thought, this logic could break. It also makes is a bit hard to follow due to the indirection.

I think we could add a function to the `models/Question.ts` to make this flow a lot easier. Maybe something like:

```ts
static getPromptType(question: IQuestion) : 'number' | 'size' | 'both' | 'none' {
  const numberPrompt = question.fields.numberPrompt
  const sizePrompt = question.fields.sizePrompt
  if (numberPrompt && sizePrompt) {
    return 'both'
  }
  if (numberPrompt) {
    return 'number'
  }
  if (sizePrompt) {
    return 'size'
  }
  return 'none'
}
```

Then, in your code here, the process should be a lot more straightforward:

```ts
const promptType = Question.getPromptType(question)
if (promptType === 'number') { return 'number' }
if (promptType === 'size' { return 'size' }
if (promptType === 'both' { throw new Error('Pre- or post-test question had two prompts instead of one.') }
if (promptType === 'none' { throw new Error('Pre- or post-test question missing prompt') }
```